akka-actor-typed/src/main/scala/akka/actor/typed/internal/delivery/WorkPullingProducerController.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal.delivery
+
+import java.util.concurrent.ThreadLocalRandom
+
+import scala.reflect.ClassTag
+
+import akka.Done
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.ActorContext
+import akka.actor.typed.scaladsl.Behaviors
+
+object WorkPullingProducerController {
+
+  sealed trait InternalCommand
+
+  sealed trait Command[A] extends InternalCommand
+
+  private final case class Next[A](sendNextTo: ActorRef[A], belongsTo: ActorRef[ConsumerController.Command[A]])
+      extends InternalCommand
+
+  private final case class OutState[A](
+      producerController: ActorRef[ProducerController.Command[A]],
+      sendNextTo: Option[ActorRef[A]])
+
+  private final case class State[A](
+      consumers: Map[ActorRef[ConsumerController.Command[A]], OutState[A]],
+      hasRequested: Boolean)
+
+  // TODO Now the workers have to be registered explicitly/manually.
+  // We could support automatic registration via Receptionist, similar to how routers work.
```

Comment:
Was just going to ask if you had thought about how to find the remote consumer, for this case it makes sense, one producer many consumers/workers. But for a 1:1 reliable delivery between cluster nodes that is potentially short lived I don't think the receptionist is a great fit (maybe I just made that use case up though, not sure) .

Reply:
I agree, Receptionist shouldn't be used for short lived.
The point-to-point scenario currently requires some application specific way to exchange CC/PC ActorRef to connect them.